include/odp/api/spec/ipsec.h

```diff
@@ -983,9 +983,55 @@ typedef struct odp_ipsec_op_flag_t {
  * These may be used to override some SA level options
  */
 typedef struct odp_ipsec_out_opt_t {
+	/** Union of all flag bits */
+	union {
+		/** Option flags. Set flag for those options that are
+		 *  used, all other options are ignored. */
+		struct {
+			/** Use fragmentation mode option */
+			uint32_t frag_mode: 1;
+
+			/** Use IP parameters option */
+			uint32_t ip_param:  1;
+
+			/** Use TFC padding length option */
+			uint32_t tfc_pad:   1;
+
+			/** Tunnel mode TFC dummy packet. In tunnel mode, set
```

Comment:
Hmm. I think RFC 4303 does not limit TFC dummy packets to tunnel mode. One can generate them in transport mode.

Reply:
That is right, but there is no limitation in this API.

This bit is just for tunnel mode dummy packets that cannot otherwise be sent. Transport mode dummy TFC packets are sent in the normal way: The input packet to an oubound IPsec operation is a well formed IP packet just like in the normal packet case, but the application just sets the protocol field of the IP header to 59. The ODP implementation needs the IP header to be there since that IP header is used (after some adjustments) in the resulting ESP/AH packet. This is different from tunnel mode where the outer IP header is generated based on the information in the SA.